tests/test_scalars.py

```diff
@@ -1236,249 +1168,211 @@ def test_timestamp_add_sub() -> None:
+    gt_dt1 = check(assert_type(ts > dt_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    le_dt1 = check(assert_type(ts <= dt_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (gt_dt1 != le_dt1).all()
+    gt_dt2 = check(assert_type(dt_idx > ts, np_1darray[np.bool]), np_1darray[np.bool])
+    le_dt2 = check(assert_type(dt_idx <= ts, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (gt_dt2 != le_dt2).all()
+    gt_un1 = check(assert_type(ts > un_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    le_un1 = check(assert_type(ts <= un_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (gt_un1 != le_un1).all()
+    gt_un2 = check(assert_type(un_idx > ts, np_1darray[np.bool]), np_1darray[np.bool])
+    le_un2 = check(assert_type(un_idx <= ts, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (gt_un2 != le_un2).all()
+
+    # <, >=
+    lt_dt1 = check(assert_type(ts < dt_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    ge_dt1 = check(assert_type(ts >= dt_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (lt_dt1 != ge_dt1).all()
+    lt_dt2 = check(assert_type(dt_idx < ts, np_1darray[np.bool]), np_1darray[np.bool])
+    ge_dt2 = check(assert_type(dt_idx >= ts, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (lt_dt2 != ge_dt2).all()
+    lt_un1 = check(assert_type(ts < un_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    ge_un1 = check(assert_type(ts >= un_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (lt_un1 != ge_un1).all()
+    lt_un2 = check(assert_type(un_idx < ts, np_1darray[np.bool]), np_1darray[np.bool])
+    ge_un2 = check(assert_type(un_idx >= ts, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (lt_un2 != ge_un2).all()
+
+    # ==, !=
+    eq_dt1 = check(assert_type(ts == dt_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    ne_dt1 = check(assert_type(ts != dt_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (eq_dt1 != ne_dt1).all()
+    eq_un1 = check(assert_type(ts == un_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    ne_un1 = check(assert_type(ts != un_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (eq_un1 != ne_un1).all()
+
+    # ==, != (ts on the rhs, use == and != of lhs)
+    eq_rhs_dt1 = check(
+        assert_type(dt_idx == ts, np_1darray[np.bool]), np_1darray[np.bool]
+    )
+    ne_rhs_dt1 = check(
+        assert_type(dt_idx != ts, np_1darray[np.bool]), np_1darray[np.bool]
+    )
+    assert (eq_rhs_dt1 != ne_rhs_dt1).all()
+    eq_rhs_un1 = check(
+        assert_type(un_idx == ts, np_1darray[np.bool]), np_1darray[np.bool]
+    )
+    ne_rhs_un1 = check(
+        assert_type(un_idx != ts, np_1darray[np.bool]), np_1darray[np.bool]
+    )
+    assert (eq_rhs_un1 != ne_rhs_un1).all()
+
+
+def test_timestamp_cmp_array() -> None:
+    ts = pd.Timestamp(year=2000, month=3, day=24, hour=12, minute=27)
+    arr_nd: npt.NDArray[np.datetime64] = np.array([1, 2, 3, 4], dtype="datetime64[ns]")
+    arr_2d = arr_nd.reshape(2, 2)
```

Comment:
Should add test for `arr_1d = pd.DatetimeIndex(["2000-1-1", 2001-2-2"]).to_numpy()` since that is 1D